compose/src/main/java/com/bottlerocketstudios/compose/snippets/snippetDetails/snippetDetailsWidgets/SnippetDetailsFileCard.kt

```diff
@@ -0,0 +1,123 @@
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.bottlerocketstudios.brarchitecture.domain.models.SnippetDetailsFile
+import com.bottlerocketstudios.compose.R
+import com.bottlerocketstudios.compose.resources.Colors
+import com.bottlerocketstudios.compose.resources.Dimens
+import com.bottlerocketstudios.compose.resources.brown_grey_three
+import com.bottlerocketstudios.compose.resources.typography
+import com.bottlerocketstudios.compose.snippets.snippetDetails.returnMockSnippetDetails
+import com.bottlerocketstudios.compose.util.convertToImageBitmap
+import com.bottlerocketstudios.compose.widgets.IconText
+import com.bottlerocketstudios.launchpad.compose.light
+import com.bottlerocketstudios.launchpad.compose.normal
+
+@Composable
+fun SnippetDetailsFilesCard(file: SnippetDetailsFile?) {
+    var expanded by remember { mutableStateOf(false) }
+
+    Card(
+        elevation = Dimens.plane_3,
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
+            .padding(vertical = Dimens.grid_1, horizontal = Dimens.grid_2)
+    ) {
+        Column(
+            modifier = Modifier.animateContentSize(tween(1000))
+        ) {
+            Row(
+                modifier = Modifier
+                    .padding(
+                        vertical = Dimens.grid_1,
+                        horizontal = Dimens.grid_2
+                    )
+                    .fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                IconText(
+                    iconRes = R.drawable.ic_file,
+                    iconColor = Colors.tertiary,
+                    text = file?.fileName ?: "",
+                    style = typography.h4.normal()
+                )
+                OutlinedButton(
+                    onClick = { expanded = !expanded },
+                    colors = ButtonDefaults.buttonColors(backgroundColor = Colors.surface),
+                    elevation = null
+                ) {
+                    Text(
+                        text = if (expanded) "Close" else "RAW",
+                        style = typography.body1.copy(color = Colors.onSurface)
+                    )
+                }
+            }
+
+            if (expanded) {
+                Divider(
+                    color = brown_grey_three,
+                    modifier = Modifier
+                        .padding(horizontal = Dimens.grid_2)
+                        .wrapContentHeight()
+                        .fillMaxWidth()
+                )
+                RawFileData(byteArray = file?.rawFile)
+            }
+        }
+    }
+}
+
+@Composable
+fun RawFileData(byteArray: ByteArray?) {
```

Comment:
@MMurrayAtBR If byteArry is null, nothing is rendered onscreen. I'd recommend making this argument non-null and  not calling the function instead of taking in a nullable type and doing nothing

cc: @BottleRocket-Colin

Reply:
Good idea